java/src/org/openqa/selenium/remote/service/DriverCommandExecutor.java

```diff
@@ -131,7 +131,7 @@ public Response execute(Command command) throws IOException {
         Thread.currentThread().interrupt();
         throw new WebDriverException("Timed out waiting for driver server to stop.", e);
       } finally {
-        executorService.shutdownNow();
+        executorService.shutdown();
```

Comment:
For the command executor, which in turn, uses the HTTP client to talk to the WebDriver, the client might have high-timeout values set, so the shutdown can take a long time if we wait for it to complete, especially if multiple-long running threads are there.  I think it might be a good idea in general to couple the shutdown() with `awaitTermination()` with a timeout. What do you think?

Reply:
@pujagani Yes, I have added `awaitTermination()` right after `shutdown()`. Thank you for the idea.